shiny/ui/_toast.py

```diff
@@ -0,0 +1,585 @@
+                header_tag = div(
+                    {"class": "toast-header"},
+                    header_content,
+                    close_button if self.closable else None,
+                )
+                contents.append(header_tag)
+
+        # Build body with optional close button
+        # * If header exists, close button goes in header
+        # * If no header, close button goes in body (if closable)
+        body_has_close_btn = self.header is None and self.closable
+
+        if not body_has_close_btn and self.icon is None:
+            # Simple body: no close button, no icon
+            body_tag = div({"class": "toast-body"}, self.body)
+        else:
+            # Complex body: has close button and/or icon
+            body_contents: list[TagChild] = []
+
+            # Add icon if present
+            if self.icon is not None:
+                body_contents.append(tags.span(self.icon, class_="toast-body-icon"))
+
+            # Add body content
+            body_contents.append(
+                div({"class": "toast-body-content flex-grow-1"}, self.body)
+            )
+
+            # Add close button if needed
+            if body_has_close_btn:
+                body_contents.append(close_button)
+
+            body_tag = div({"class": "toast-body d-flex gap-2"}, *body_contents)
+
+        contents.append(body_tag)
+
+        # Build final toast tag
+        return div(
+            {
+                "class": " ".join(classes),
+                # Override id if provided, or generate a random ID if missing
+                "id": id or self.id or _toast_random_id(),
+            },
+            *contents,
+            components_dependencies(),
+            role=role,
+            aria_live=aria_live,
+            aria_atomic="true",
+            data_bs_autohide="true" if self.autohide else "false",
+            **self.attribs,
+        )
+
+
+@add_example()
+def toast(
+    *args: TagChild | TagAttrs,
+    header: Optional[str | ToastHeader | TagChild] = None,
+    icon: Optional[TagChild] = None,
+    id: Optional[str] = None,
+    type: Optional[
+        Literal[
+            "primary",
+            "secondary",
+            "success",
+            "info",
+            "warning",
+            "danger",
+            "error",
+            "light",
+            "dark",
+        ]
+    ] = None,
+    duration_s: Optional[int | float] = 5,
+    position: str | list[str] | tuple[str, ...] = "top-right",
+    closable: bool = True,
+    **kwargs: TagAttrValue,
+) -> Toast:
+    """
+    Create a toast notification object.
+
+    Toast notifications are temporary, non-intrusive messages that appear on screen to
+    provide feedback to users. They support multiple semantic types, flexible
+    positioning, auto-hide with progress bars, and optional headers with icons.
+
+    Parameters
+    ----------
+    *args
+        Body content (HTML elements or strings). Named arguments become HTML attributes.
```

Comment:
Named arguments are `kwargs`


```suggestion
        Body content (HTML elements or strings).
```